crescent/plugin.py

```diff
@@ -38,7 +39,23 @@ def unload(self, path: str) -> None:
         plugin = self.plugins.pop(path)
         plugin._unload()
 
-    def load(self, path: str, refresh: bool = False) -> Plugin:
+    @overload
+    def load(self, path: str, /, *, refresh: bool = ...) -> Plugin:
+        ...
+
+    @overload
+    def load(self, path: str, *, strict: Literal[True], refresh: bool = ...) -> Plugin:
+        ...
+
+    @overload
+    def load(self, path: str, *, strict: Literal[False], refresh: bool = ...) -> Plugin | None:
+        ...
+
+    @overload
+    def load(self, path: str, refresh: bool = ..., strict: bool = ...) -> Plugin | None:
+        ...
+
+    def load(self, path: str, refresh: bool = False, strict: bool = True) -> Plugin | None:
```

Comment:
Can't this just be
```py
@overload
def load(self, path: str, refresh: bool = ..., strict: Literal[True] = ...) -> Plugin:
    ...

@overload
def load(self, path: str, refresh: bool = ..., strict: Literal[False] = False) -> Plugin | None:
    ...

def load(self, path: str, refresh: bool = False, strict: bool = True) -> Plugin | None:
    ...
```

Reply:
No, `bot.plugins.load(mod_name, strict=a_variable)` has the no overload variants match error.

Reply:
Try
```py
@overload
def load(self, path: str, refresh: bool = ..., strict: Literal[True] = ...) -> Plugin:
    ...

@overload
def load(self, path: str, refresh: bool = ..., strict: Literal[False]) -> Plugin | None:
    ...

def load(self, path: str, refresh: bool = False, strict: bool = True) -> Plugin | None:
    ...
```

Reply:
This overload doesn't work. If i fix it with a star in the second one I get the same error as the first overload.